lib/broadside/deploy/ecs_deploy.rb

```diff
@@ -150,141 +154,96 @@ def bash
+        w.delay = config.ecs.poll_frequency
+        seen_event = nil
+
+        w.before_wait do |attempt, response|
+          debug "(#{attempt}/#{w.max_attempts}) Polling ECS for events..."
+          # skip first event since it doesn't apply to current request
+          if response.services[0].events.first && response.services[0].events.first.id != seen_event && attempt > 1
+            seen_event = response.services[0].events.first.id
+            debug(response.services[0].events.first.message)
           end
-        rescue Aws::Waiters::Errors::TooManyAttemptsError
-          exception 'Deploy did not finish in the expected amount of time.'
         end
-      else
-        exception 'Failed to update service during deploy.'
       end
     end
 
     def run_command(command)
       command_name = command.join(' ')
-      resp = ecs_client.run_task({
-        cluster: config.ecs.cluster,
-        task_definition: get_latest_task_def_id,
-        overrides: {
-          container_overrides: [
-            {
-              name: family,
-              command: command
-            },
-          ],
-        },
-        count: 1,
-        started_by: "before_deploy:#{command_name}"[0...36]
-      })
-
-      if resp.successful?
-        task_id = resp.tasks[0].task_arn
-        debug "Launched #{command_name} task #{task_id}"
-        debug "Waiting for #{command_name} to complete..."
-        ecs_client.wait_until(:tasks_stopped, {cluster: config.ecs.cluster, tasks: [task_id]}) do |w|
-          w.max_attempts = nil
-          w.delay = config.ecs.poll_frequency
-          w.before_attempt do |attempt|
-            debug "Attempt #{attempt}: waiting for #{command_name} to complete..."
-          end
-        end
-        debug 'Task finished running, getting logs...'
-        info "#{command_name} task container logs:\n#{get_container_logs(task_id)}"
-        if (code = get_task_exit_code(task_id)) == 0
-          debug "#{command_name} task #{task_id} exited with status code 0"
-        else
-          exception "#{command_name} task #{task_id} exited with a non-zero status code #{code}!"
+      run_task_response = EcsManager.run_task(config.ecs.cluster, family, command)
+
+      unless run_task_response.successful?
+        exception("Failed to run #{command_name} task.", run_task_response.pretty_inspect)
+      end
+
+      task_arn = run_task_response.tasks[0].task_arn
+      debug "Launched #{command_name} task #{task_arn}, waiting for completion..."
+
+      EcsManager.ecs.wait_until(:tasks_stopped, { cluster: config.ecs.cluster, tasks: [task_arn] }) do |w|
+        w.max_attempts = nil
+        w.delay = config.ecs.poll_frequency
+        w.before_attempt do |attempt|
+          debug "Attempt #{attempt}: waiting for #{command_name} to complete..."
         end
+      end
+
+      info "#{command_name} task container logs:\n#{get_container_logs(task_arn)}"
+
+      # TODO: it should check ALL the task exit codes from all the runs
```

Comment:
yeah i guess you are write; seems like it only actually runs on one container.  i'll delete this TODO